addons/fleet/models/fleet_vehicle_log_contract.py

```diff
@@ -83,6 +83,21 @@ def _compute_days_left(self):
 
     def write(self, vals):
         res = super(FleetVehicleLogContract, self).write(vals)
+        if 'start_date' in vals or 'expiration_date' in vals:
+            date_today = fields.Date.from_string(fields.Date.today())
+            futur_contracts = self.env[self._name]
+            running_contracts = self.env[self._name]
+            expired_contracts = self.env[self._name]
+            for contract in self.filtered(lambda c: c.start_date and c.state != 'closed'):
+                if date_today < contract.start_date:
+                    futur_contracts |= contract
+                elif not contract.expiration_date or contract.start_date < date_today < contract.expiration_date:
```

Comment:
Shouldn't we do something like this to avoid setting the contract as expired if the start_date is set to today ? 
```suggestion
                elif not contract.expiration_date or contract.start_date <= date_today < contract.expiration_date:```